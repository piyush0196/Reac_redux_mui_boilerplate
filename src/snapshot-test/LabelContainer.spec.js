import TestRenderer from 'react-test-renderer'
import LabelContainer from 'components/ui-kit/LabelContainer'
import { ThemeProvider } from '@mui/material'
import { theme } from 'theme/theme'

describe('Component - LabelContainer', () => {
  const wrappers = []
  const LabelContainerJSX = ({ children, type, label }) => {
    return (
      <ThemeProvider theme={theme}>
        <LabelContainer type={type} label={label}>
          {children}
        </LabelContainer>
      </ThemeProvider>
    )
  }

  const types = ['filter', 'default', 'dateFilter']

  types.forEach((type) =>
    wrappers.push(
      TestRenderer.create(
        <LabelContainerJSX type={type} label='label'>
          <div>Label text</div>
        </LabelContainerJSX>
      )
    )
  )
  wrappers.push(
    TestRenderer.create(
      <LabelContainerJSX label='label'>
        <div>Label text</div>
      </LabelContainerJSX>
    )
  )

  it('renders all labels correctly', () => {
    wrappers.forEach((wrapper) => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
