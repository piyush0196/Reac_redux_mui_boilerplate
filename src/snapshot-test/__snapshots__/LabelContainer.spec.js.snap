// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - LabelContainer renders all labels correctly 1`] = `
<div
  className="MuiBox-root css-1cu9w86"
  type="filter"
>
  <div
    style={
      Object {
        "marginRight": "10px",
        "width": "max-content",
      }
    }
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-5ts02r-MuiTypography-root"
    >
      label
    </h6>
  </div>
  <div>
    Label text
  </div>
</div>
`;

exports[`Component - LabelContainer renders all labels correctly 2`] = `
<div
  style={
    Object {
      "height": "100px",
      "width": "fit-content",
    }
  }
>
  <div
    style={
      Object {
        "textAlign": "start",
      }
    }
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-5ts02r-MuiTypography-root"
    >
      label*
    </h6>
  </div>
  <div>
    Label text
  </div>
</div>
`;

exports[`Component - LabelContainer renders all labels correctly 3`] = `
<div
  className="MuiBox-root css-i9rics"
  type="dateFilter"
>
  <div
    style={
      Object {
        "marginRight": "10px",
        "width": "max-content",
      }
    }
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-5ts02r-MuiTypography-root"
    >
      label
    </h6>
  </div>
  <div>
    Label text
  </div>
</div>
`;

exports[`Component - LabelContainer renders all labels correctly 4`] = `
<div
  style={
    Object {
      "height": "100px",
      "width": "fit-content",
    }
  }
>
  <div
    style={
      Object {
        "textAlign": "start",
      }
    }
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-5ts02r-MuiTypography-root"
    >
      label*
    </h6>
  </div>
  <div>
    Label text
  </div>
</div>
`;
